dist/screeps.d.ts

```diff
@@ -2,69 +2,308 @@
+declare var STRUCTURE_POWER_SPAWN: StructureConstPowerSpawn;
+declare var STRUCTURE_EXTRACTOR: StructureConstExtractor;
+declare var STRUCTURE_LAB: StructureConstLab;
+declare var STRUCTURE_TERMINAL: StructureConstTerminal;
+declare var STRUCTURE_CONTAINER: StructureConstContainer;
+declare var STRUCTURE_NUKER: StructureConstNuker;
+declare type ObjectConst = StructureConst | "creep" | "wall";
+declare type ResourceConst = ResourceConstEnergy | ResourceConstPower | ResourceConstH | ResourceConstO | ResourceConstU | ResourceConstL | ResourceConstK | ResourceConstZ | ResourceConstX | ResourceConstG | ResourceConstOH | ResourceConstZK | ResourceConstUL | ResourceConstUH | ResourceConstUO | ResourceConstKH | ResourceConstKO | ResourceConstLH | ResourceConstLO | ResourceConstZH | ResourceConstZO | ResourceConstGH | ResourceConstGO | ResourceConstUH2O | ResourceConstUHO2 | ResourceConstKH2O | ResourceConstKHO2 | ResourceConstLH2O | ResourceConstLHO2 | ResourceConstZH2O | ResourceConstZHO2 | ResourceConstGH2O | ResourceConstGHO2 | ResourceConstXUH2O | ResourceConstXUHO2 | ResourceConstXKH2O | ResourceConstXKHO2 | ResourceConstXLH2O | ResourceConstXLHO2 | ResourceConstXZH2O | ResourceConstXZHO2 | ResourceConstXGH2O | ResourceConstXGHO2;
+declare type ResourceConstEnergy = "energy";
+declare type ResourceConstPower = "power";
+declare type ResourceConstH = "H";
+declare type ResourceConstO = "O";
+declare type ResourceConstU = "U";
+declare type ResourceConstL = "L";
+declare type ResourceConstK = "K";
+declare type ResourceConstZ = "Z";
+declare type ResourceConstX = "X";
+declare type ResourceConstG = "G";
+declare type ResourceConstOH = "OH";
+declare type ResourceConstZK = "ZK";
+declare type ResourceConstUL = "UL";
+declare type ResourceConstUH = "UH";
+declare type ResourceConstUO = "UO";
+declare type ResourceConstKH = "KH";
+declare type ResourceConstKO = "KO";
+declare type ResourceConstLH = "LH";
+declare type ResourceConstLO = "LO";
+declare type ResourceConstZH = "ZH";
+declare type ResourceConstZO = "ZO";
+declare type ResourceConstGH = "GH";
+declare type ResourceConstGO = "GO";
+declare type ResourceConstUH2O = "UH2O";
+declare type ResourceConstUHO2 = "UHO2";
+declare type ResourceConstKH2O = "KH2O";
+declare type ResourceConstKHO2 = "KHO2";
+declare type ResourceConstLH2O = "LH2O";
+declare type ResourceConstLHO2 = "LHO2";
+declare type ResourceConstZH2O = "ZH2O";
+declare type ResourceConstZHO2 = "ZHO2";
+declare type ResourceConstGH2O = "GH2O";
+declare type ResourceConstGHO2 = "GHO2";
+declare type ResourceConstXUH2O = "XUH2O";
+declare type ResourceConstXUHO2 = "XUHO2";
+declare type ResourceConstXKH2O = "XKH2O";
+declare type ResourceConstXKHO2 = "XKHO2";
+declare type ResourceConstXLH2O = "XLH2O";
+declare type ResourceConstXLHO2 = "XLHO2";
+declare type ResourceConstXZH2O = "XZH2O";
+declare type ResourceConstXZHO2 = "XZHO2";
+declare type ResourceConstXGH2O = "XGH2O";
+declare type ResourceConstXGHO2 = "XGHO2";
```

Comment:
Is one type per resource actually what we want? Don't we instead want a string literal type?

Something like:

```
declare type ResourceType = 
    "energy" |
    "H" |
    "L" |
    "X" |
    ...
```